pgx/src/datum/date.rs

```diff
@@ -98,24 +98,54 @@ impl Date {
 }
 
 #[cfg(feature = "time-crate")]
-impl From<time::Date> for Date {
-    #[inline]
-    fn from(date: time::Date) -> Self {
-        Date::from_pg_epoch_days(date.to_julian_day() - POSTGRES_EPOCH_JDATE)
+mod with_time_crate {
+    use crate::{Date, POSTGRES_EPOCH_JDATE};
+    use core::fmt::{Display, Formatter};
+    use std::error::Error;
+
+    #[derive(Debug, PartialEq, Clone)]
+    #[non_exhaustive]
+    pub struct TryFromDateError(pub Date);
```

Comment:
Do we want to export this in a way users can access it?

Reply:
I don’t see that they’ll need to reference it by name.

Reply:
It’s under this time-crate feature flag anyways. And honestly, I hope that’ll go away soon

Reply:
alrighty.  done